pkg/storage/backend-storage/backend-storage.go

```diff
@@ -126,12 +232,12 @@ func CreateIfNeeded(vmi *corev1.VirtualMachineInstance, clusterConfig *virtconfi
 // - No PVC is needed for the VMI since it doesn't use backend storage
 // - The backend storage PVC is bound
 // - The backend storage PVC is pending uses a WaitForFirstConsumer storage class
-func IsPVCReady(vmi *corev1.VirtualMachineInstance, client kubecli.KubevirtClient, scStore cache.Store) (bool, error) {
+func (bs *BackendStorage) IsPVCReady(vmi *corev1.VirtualMachineInstance) (bool, error) {
 	if !IsBackendStorageNeededForVMI(&vmi.Spec) {
 		return true, nil
 	}
 
-	pvc, err := client.CoreV1().PersistentVolumeClaims(vmi.Namespace).Get(context.Background(), PVCForVMI(vmi), metav1.GetOptions{})
+	pvc, err := bs.client.CoreV1().PersistentVolumeClaims(vmi.Namespace).Get(context.Background(), PVCForVMI(vmi), metav1.GetOptions{})
```

Comment:
Can we use the informer here as well?

Reply:
This was previously discussed, but I can't find the conversation anymore...
`IsPVCReady()` is called right after PVC creation, and I'm pretty sure the informer won't yet know about the new object in most (all?) cases.
That means the VMI controller always fails a `sync()` (and pushes the VMI key to the back of the queue) on first creation of a VMI with backend storage, which feels more expensive than an API call.
Let me know if you still think using the informer makes sense here and I'll change it.

Reply:
I think the `Get` in `CreateIfNeededAndUpdateVolumeStatus` could be be pretty expensive because that is called whenever a VMI is reconciled

`IsPVCReady` is called less frequently but I see your point.  It may be called right after the PVC is created.  

An ideal solution may be for `CreateIfNeededAndUpdateVolumeStatus` to return a flag when it actually creates the PVC and then the VMI controller can set an expectation and exit the reconcile loop.

If that is something you don't want to tackle now maybe just use the informer for `CreateIfNeededAndUpdateVolumeStatus`?

Reply:
Yeah that makes sense. I'll switch to the informer in `CreateIfNeededAndUpdateVolumeStatus`.